nxcomp/src/Loop.cpp

```diff
@@ -11688,43 +11688,44 @@ int SetPorts()
   // ing, causing a loop.
   //
 
-  useCupsSocket = 0;
-  if (cupsPort.enabled()) {
-    if (control -> ProxyMode == proxy_client) {
-      cupsPort.setDefaultTCPPort(DEFAULT_NX_CUPS_PORT_OFFSET + proxyPort);
-      useCupsSocket = 1;
-    }
-    else
-      cupsPort.setDefaultTCPPort(631);
+  if (control -> ProxyMode == proxy_client) {
+    // ChannelEndPoint::enabled() implements the logic described above,
+    // and takes the default port into consideration. If cups=1, and
+    // there is a default port, then enabled() will return true.
+    //
+    // Therefore, we must set the default port before calling this
+    // function.
+    cupsPort.setDefaultTCPPort(DEFAULT_NX_CUPS_PORT_OFFSET + proxyPort);
+    useCupsSocket = cupsPort.enabled();
+  } else {
+    cupsPort.setDefaultTCPPort(631);
   }
 
+
+
 #ifdef TEST
   *logofs << "Loop: cups port: " << cupsPort << "\n"
             << logofs_flush;
 #endif
 
-  useAuxSocket = 0;
-  if (auxPort.enabled()) {
-    if (control -> ProxyMode == proxy_client) {
-      auxPort.setDefaultTCPPort(DEFAULT_NX_AUX_PORT_OFFSET + proxyPort);
-      useAuxSocket = 1;
-    }
-    else {
-      auxPort.setDefaultTCPPort(1);
-
-      if (auxPort.getTCPPort() != 1) {
+  // TODO: FIXME
```

Comment:
Why FIXME? Please be more verbose here on the issue that you still see unresolved here.

Reply:
Er, oops. That should have been removed, the fix was already made.